### PR TITLE
Automation change mode dialog

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add change mode action in overflow menu
Add alert for disabled state
I also fixed some lint issues in rename dialog
The name was duplicated in yaml mode, I removed it.

### Overflow menu
Capture d’écran 2022-09-05 à 12 31 36

### Change mode dialog 
Capture d’écran 2022-09-05 à 12 31 29

### Disable state (YAML)
Capture d’écran 2022-09-05 à 15 35 18

### Disable state (GUI)
Capture d’écran 2022-09-05 à 15 35 11

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    this.closeDialog();
+  }
+
+  static get styles(): CSSResultGroup {
+    return [
+      haStyle,
+      haStyleDialog,
+      css`
+        ha-textfield,
+        ha-textarea {
```
Comment (bramkragten) [inline src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts:147]:
```suggestion
        ha-textarea,
        ha-select {
```

Comment (bramkragten) [inline src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts:147]:
```suggestion
        ha-select {
```

File: src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        </ha-select>
+        ${this._newMode && isMaxMode(this._newMode)
+          ? html`
+              <br /><ha-textfield
+                .label=${this.hass.localize(
+                  `ui.panel.config.automation.editor.max.${this._newMode}`
+                )}
+                type="number"
+                name="max"
+                .value=${this._newMax || "10"}
```
Comment (bramkragten) [inline src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts:98]:
```suggestion
                .value=${this._newMax}
```

File: src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      </ha-dialog>
+    `;
+  }
+
+  private _modeChanged(ev) {
+    const mode = ev.target.value;
+    this._newMode = mode;
+    if (!isMaxMode(mode)) {
+      this._newMax = undefined;
+    }
```
Comment (bramkragten) [inline src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts:121]:
```suggestion
    if (!isMaxMode(mode)) {
      this._newMax = undefined;
    } else if (!this._newMax) {
    	this._newMax = 10;
    }
```

File: src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
+import "@material/mwc-button";
+import { css, CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { createCloseHeading } from "../../../../components/ha-dialog";
+import "../../../../components/ha-textfield";
```
Comment (bramkragten) [inline src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts:6]:
```suggestion
```

File: src/panels/config/automation/ha-automation-editor.ts [hunk 5]
```diff
@@ -161,11 +163,26 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
...
+          <mwc-list-item
+            graphic="icon"
+            @click=${this._promptAutomationMode}
+            .disabled=${!this.automationId || this._mode === "yaml"}
+          >
+            ${this.hass.localize(
+              "ui.panel.config.automation.editor.change_mode"
+            )}
+            <ha-svg-icon slot="graphic" .path=${mdiDebugStepOver}></ha-svg-icon>
+          </mwc-list-item>
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:184]:
```suggestion
          ${this. _config.use_blueprint ? "" : html`<mwc-list-item
            graphic="icon"
            @click=${this._promptAutomationMode}
            .disabled=${!this.automationId || this._mode === "yaml"}
          >
            ${this.hass.localize(
              "ui.panel.config.automation.editor.change_mode"
            )}
            <ha-svg-icon slot="graphic" .path=${mdiDebugStepOver}></ha-svg-icon>
          </mwc-list-item>`}
```

File: src/panels/config/automation/ha-automation-editor.ts [hunk 6]
```diff
@@ -243,6 +260,20 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
                 })}"
                 @subscribe-automation-config=${this._subscribeAutomationConfig}
               >
+                ${!stateObj || stateObj.state === "off"
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:263]:
```suggestion
                ${stateObj?.state === "off"
```

File: src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts (hunk 4 above)
Comment (piitaya) [reply src/panels/config/automation/automation-mode-dialog/dialog-automation-mode.ts:6]: ha-text-field is needed for max input

File: src/panels/config/automation/ha-automation-editor.ts (hunk 5 above)
Comment (piitaya) [reply src/panels/config/automation/ha-automation-editor.ts:184]: 👍 I forgot this condition 🙂

**Review by bramkragten — APPROVED** [review]

**Pull request merged**